### PR TITLE
optimizing Trace log for subject

### PR DESCRIPTION
**Pull request by maeste**

### CODE REVIEW

File: core/src/main/java/org/jboss/jca/core/connectionmanager/pool/AbstractPool.java [hunk 1]
```diff
@@ -1044,6 +1044,8 @@ public PoolStatisticsImpl getInternalStatistics()
    protected boolean internalTestConnection(ConnectionRequestInfo cri, Subject subject)
    {
       log.debugf("%s: testConnection(%s, %s)", poolName, cri, subject);
+      log.debugf("testConnection subjectIdentityHashCode=%s", System.identityHashCode(subject));
```
Comment (jesperpedersen) [inline core/src/main/java/org/jboss/jca/core/connectionmanager/pool/AbstractPool.java]: I think we can just add it to the previous line (%s, %s) (%d) or something

File: core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/ManagedConnectionPoolUtility.java [hunk 2]
```diff
@@ -80,13 +82,15 @@ static String details(String method, String poolName, int inUse, int max)
...
                              Collection<ConnectionListener> available, Collection<ConnectionListener> inUse,
-                             PoolStatisticsImpl ps)
+                             PoolStatisticsImpl ps, Subject subject, ConnectionRequestInfo cri)
    {
       StringBuilder sb = new StringBuilder(1024);
       long now = System.currentTimeMillis();
 
       sb.append(method).append(newLine);
       sb.append("Method: ").append(method).append(newLine);
+      sb.append("Subject identityHashCode: ").append(System.identityHashCode(subject)).append(newLine);
```
Comment (jesperpedersen) [inline core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/ManagedConnectionPoolUtility.java]: Missing 2 prefix spaces, and just do "Subject: "

File: core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/ManagedConnectionPoolUtility.java [hunk 3]
```diff
@@ -80,13 +82,15 @@ static String details(String method, String poolName, int inUse, int max)
...
+      sb.append("CRI identityHashCode: ").append(System.identityHashCode(cri)).append(newLine);
```
Comment (jesperpedersen) [inline core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/ManagedConnectionPoolUtility.java]: Same deal here

File: core/src/main/java/org/jboss/jca/core/security/picketbox/PicketBoxSubjectFactory.java [hunk 4]
```diff
@@ -56,7 +63,14 @@ public Subject createSubject()
     */
    public Subject createSubject(String sd)
    {
-      return delegator.createSubject(sd);
+      Subject subject = delegator.createSubject(sd);
+
+      if (log.isTraceEnabled())
```
Comment (jesperpedersen) [inline core/src/main/java/org/jboss/jca/core/security/picketbox/PicketBoxSubjectFactory.java]: Better do this as a static variable

File: core/src/main/java/org/jboss/jca/core/security/picketbox/PicketBoxSubjectFactory.java [hunk 5]
```diff
@@ -56,7 +63,14 @@ public Subject createSubject()
...
+      {
+         log.trace("Subject=" + subject);
+         log.trace("Subject identity=" + System.identityHashCode(subject));
```
Comment (jesperpedersen) [inline core/src/main/java/org/jboss/jca/core/security/picketbox/PicketBoxSubjectFactory.java]: And Integer.toHexString

**Comment by maeste** [comment]
Everything fixed and squashed

File: core/src/main/java/org/jboss/jca/core/connectionmanager/pool/AbstractPool.java [hunk 6]
```diff
@@ -1043,7 +1043,8 @@ public PoolStatisticsImpl getInternalStatistics()
     */
    protected boolean internalTestConnection(ConnectionRequestInfo cri, Subject subject)
    {
-      log.debugf("%s: testConnection(%s, %s)", poolName, cri, subject);
+      log.debugf("%s: testConnection(%s, %s)(%d)", poolName, cri, subject, System.identityHashCode(subject));
```
Comment (jesperpedersen) [inline core/src/main/java/org/jboss/jca/core/connectionmanager/pool/AbstractPool.java]: %s + Integer.toHexString()

File: core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/ManagedConnectionPoolUtility.java [hunk 7]
```diff
@@ -80,13 +82,15 @@ static String details(String method, String poolName, int inUse, int max)
...
                              Collection<ConnectionListener> available, Collection<ConnectionListener> inUse,
-                             PoolStatisticsImpl ps)
+                             PoolStatisticsImpl ps, Subject subject, ConnectionRequestInfo cri)
    {
       StringBuilder sb = new StringBuilder(1024);
       long now = System.currentTimeMillis();
 
       sb.append(method).append(newLine);
       sb.append("Method: ").append(method).append(newLine);
+      sb.append("  Subject: ").append(Integer.toHexString(System.identityHashCode(subject))).append(newLine);
```
Comment (jesperpedersen) [inline core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/ManagedConnectionPoolUtility.java]: Better do subject == null ? "null" : other

File: core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/ManagedConnectionPoolUtility.java [hunk 8]
```diff
@@ -80,13 +82,15 @@ static String details(String method, String poolName, int inUse, int max)
...
+      sb.append("  CRI: ").append(Integer.toHexString(System.identityHashCode(cri))).append(newLine);
```
Comment (jesperpedersen) [inline core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/ManagedConnectionPoolUtility.java]: Same here

File: core/src/main/java/org/jboss/jca/core/security/picketbox/PicketBoxSubjectFactory.java [hunk 9]
```diff
@@ -34,6 +37,12 @@
    /** Delegator */
    private org.jboss.security.SubjectFactory delegator;
 
+   /** The logger */
+   private static CoreLogger log = Logger.getMessageLogger(CoreLogger.class, PicketBoxSubjectFactory.class.getName());
+
+   private static boolean traceEnabled = log.isTraceEnabled();
```
Comment (jesperpedersen) [inline core/src/main/java/org/jboss/jca/core/security/picketbox/PicketBoxSubjectFactory.java]: We do "trace" by standard

**Comment by jesperpedersen** [comment]
Merged

**Pull request closed without merging**